tests/test_dfa.py

```diff
@@ -188,99 +269,42 @@ def test_read_input_step(self) -> None:
 
     def test_operations_other_types(self) -> None:
         """Should raise TypeError for all but equals."""
-        # This DFA accepts all words which do not contain two
-        # consecutive occurrences of 1
-        dfa = DFA(
-            states={"q0", "q1", "q2"},
-            input_symbols={"0", "1"},
-            transitions={
-                "q0": {"0": "q0", "1": "q1"},
-                "q1": {"0": "q0", "1": "q2"},
-                "q2": {"0": "q2", "1": "q2"},
-            },
-            initial_state="q0",
-            final_states={"q0", "q1"},
-        )
         other = 42
-        self.assertNotEqual(dfa, other)
+        self.assertNotEqual(self.dfa, other)
         with self.assertRaises(TypeError):
-            dfa | other  # type: ignore
+            self.dfa | other  # type: ignore
         with self.assertRaises(TypeError):
-            dfa & other  # type: ignore
+            self.dfa & other  # type: ignore
         with self.assertRaises(TypeError):
-            dfa - other  # type: ignore
+            self.dfa - other  # type: ignore
         with self.assertRaises(TypeError):
-            dfa ^ other  # type: ignore
+            self.dfa ^ other  # type: ignore
         with self.assertRaises(TypeError):
-            dfa < other  # type: ignore
+            self.dfa < other  # type: ignore
         with self.assertRaises(TypeError):
-            dfa <= other  # type: ignore
+            self.dfa <= other  # type: ignore
         with self.assertRaises(TypeError):
-            dfa > other  # type: ignore
+            self.dfa > other  # type: ignore
         with self.assertRaises(TypeError):
-            dfa >= other  # type: ignore
+            self.dfa >= other  # type: ignore
 
     def test_equivalence_not_equal(self) -> None:
```

Comment:
This test should test the partial DFAs (applies to some other tests too). I was doing something like this:

```
def partial_combinations(*args):
    args = tuple(dfa.copy() for dfa in args)
    for partial_flags in product((False, True), repeat=len(args)):
        yield partial_flags, (
            to_partial(dfa, partial_flag)
            for partial_flag, dfa in zip(partial_flags, args)
        )
```

Reply:
Then for each test I would just pass the DFAs to the `partial_combinations` function and apply the asserts normally

Reply:
This is subsumed by the `test_set_laws` function, where the equality function is used on a mix of partial and complete DFAs.